From: Andrew Morton <akpm@linux-foundation.org>
Subject: selftests-complete-kselftest-include-centralization-fix
Date: Fri Oct 17 02:27:53 PM PDT 2025

fix samples/vfs/test-list-all-mounts.c build

In file included from /tmp/next/build/samples/vfs/test-list-all-mounts.c:11:
/tmp/next/build/samples/vfs/../../tools/testing/selftests/pidfd/pidfd.h:28:10: fatal error: kselftest.h: No such file or directory
   28 | #include "kselftest.h"
      |          ^~~~~~~~~~~~~


Reported-by: Mark Brown <broonie@kernel.org>
Closes: https://lkml.kernel.org/r/aPJD4ANRWUDVDqKc@sirena.org.uk
Cc: Bala-Vignesh-Reddy <reddybalavignesh9979@gmail.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 samples/vfs/Makefile |    2 ++
 1 file changed, 2 insertions(+)

--- a/samples/vfs/Makefile~selftests-complete-kselftest-include-centralization-fix
+++ a/samples/vfs/Makefile
@@ -1,4 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0-only
 userprogs-always-y += test-fsmount test-statx mountinfo test-list-all-mounts
 
+CFLAGS += -I../../tools/testing/selftests
+
 userccflags += -I usr/include
_
